drivers/timer/renesas_rx_cmt.c

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zephyr/sys_clock.h"
+#include <zephyr/arch/cpu.h>
+#include <zephyr/init.h>
+#include <soc.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/irq.h>
+#include <platform.h>
+
+#define DT_DRV_COMPAT renesas_rx_timer_cmt
+
+#define CMT_NODE  DT_NODELABEL(cmt)
+#define CMT0_NODE DT_NODELABEL(cmt0)
+
+#define CMT_CMSTR0_ADDR DT_REG_ADDR_BY_NAME(CMT_NODE, CMSTR0)
+#define CMT0_CMCR_ADDR  DT_REG_ADDR_BY_NAME(CMT0_NODE, CMCR)
+#define CMT0_CMCNT_ADDR DT_REG_ADDR_BY_NAME(CMT0_NODE, CMCNT)
+#define CMT0_CMCOR_ADDR DT_REG_ADDR_BY_NAME(CMT0_NODE, CMCOR)
+
+#define ICU_NODE DT_NODELABEL(icu)
+
+#define ICU_IR_ADDR  DT_REG_ADDR_BY_NAME(ICU_NODE, IR)
+#define ICU_IER_ADDR DT_REG_ADDR_BY_NAME(ICU_NODE, IER)
+#define ICU_IPR_ADDR DT_REG_ADDR_BY_NAME(ICU_NODE, IPR)
+#define ICU_FIR_ADDR DT_REG_ADDR_BY_NAME(ICU_NODE, FIR)
+
+#define COUNTER_MAX   0x0000ffff
+#define TIMER_STOPPED 0xff000000
+
+/* The following macro values will be changed later to be calculated using the clock control driver.
+ * This macro is the CMT count clock. (=PCLKB/8)
+ */
+#define CYCLES_PER_SEC  (CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC)
+#define TICKS_PER_SEC   (CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+#define CYCLES_PER_TICK (CYCLES_PER_SEC / TICKS_PER_SEC)
+#define MAX_TICKS       ((COUNTER_MAX / CYCLES_PER_TICK) - 1)
+
+#if (IS_ENABLED(CONFIG_TICKLESS_KERNEL))
+#warning Warning: Tickless mode is not yet implemented.
```

Comment:
if it's not implemented then why is this a warning and not an error?

Reply:
It's changed to #error